plugin/kubernetes/parse.go

```diff
@@ -33,7 +33,9 @@ func parseRequest(state request.Request) (r recordRequest, err error) {
 	// 3. (service): service.namespace.pod|svc.zone
 	//
 	// Federations are handled in the federation plugin. And aren't parsed here.
-
+	if !dns.IsSubDomain(state.Zone, state.Name()) {
```

Comment:
I dont think this can happen.  `state.Zone` is taken from the query name in Kubernetes.ServeDNS,...

```
	zone := plugin.Zones(k.Zones).Matches(state.Name())
	if zone == "" {
		return plugin.NextOrFailure(k.Name(), k.Next, ctx, w, r)
	}

	state.Zone = zone
```

So this check is not necessary.

Reply:
The recursive CNAME lookup will not call `Kubernetes.ServeDNS` function. Just call `Kubernetes.Services`. So this zone will not checked. When use a query which cname to external name will not be checked.

Reply:
Ah, right, but the vast majority of the time it's redundant...  Perhaps move this to the back end.

Reply:
Move this check to Kubernetes.Service?

Reply:
I meant backend_lookup.go, to check before recursing.  e.g. if the CNAME target is not in the state.Zone, dont try to recurse in the current plugin. 

Do tests fail if you remove this check?

Reply:
Yes, If I move the check to backend_lookup.go, Will break the etcd testcases. 

```
=== RUN   TestOtherLookup
--- FAIL: TestOtherLookup (0.04s)
	helpers.go:131: Additional for "a.mail.skydns.test." contained 0 results, 2 expected
	helpers.go:301: ;; opcode: QUERY, status: NOERROR, id: 38075
		;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
		
		;; QUESTION SECTION:
		;a.mail.skydns.test.	IN	 MX
		
		;; ANSWER SECTION:
		a.mail.skydns.test.	300	IN	MX	50 mx.skydns.test.
```